text/0000-range_redesign.md

```diff
@@ -0,0 +1,67 @@
+- Feature Name: range_redesign
+- Start Date: 31/03/2019
+- RFC PR: (leave this empty)
+- Pony Issue: (leave this empty)
+
+# Summary
+
+A Range reimplementation that includes the features from collections.Reverse together with a more comprehensive and coherent API. A reference implementation can be found here: https://github.com/chobeat/pony-range-proposal
+
+# Motivation
+
+The existence of Range and Reverse as separate classes, each one with their own limitations, to cover a use case that in most languages is addressed by a single API highlights the possibility for an improvement. Having a single, well-tested implementation to generate numeric ranges will improve the quality of life of new developers and possibly reduce unexpected behaviors for a core piece of the stdlib.
+For example in the existing implementation it is possible to generate infinite ranges without any explicit control. 
+
+In addition to that, the current implementation doesn't allow to express a range spanning over a whole data type since it makes assumptions on the exclusiveness of the right bound.
+
+# Detailed design
+
+The new implementation should fullfil the following design principles: 
+
+* be able to generate all the possible ranges, increasing and decreasing, for all the data types, without incurring in overflow errors.
+* offer a unified, usable, coherent API with the possibility to specify inclusive or exclusive bounds. To achieve this, the definition of the bounds and the range should be independent. The Range class should offer helper methods to cover the most common cases without explicitely instancing the bounds.
+* never raise errors but instead return an empty list. This is done to make the class more usable in day to day usage. A version of the class with explicit errors can be considered.
```

Comment:
can you elaborate on why this is a design goal?

Reply:
because it's less verbose for something that shouldn't have runtime errors when the use defines the range with known numbers. When you're using runtime values, after the definition of the range you can know if it's valid or not and handle both cases anyway by checking this condition.

Reply:
Maybe I will add an example to the RFC comparing the two versions